crypto/x509/v3_utl.c

```diff
@@ -60,6 +64,14 @@ static int x509v3_add_len_value(const char *name, const char *value,
         ERR_raise(ERR_LIB_X509V3, ERR_R_CRYPTO_LIB);
         goto err;
     }
+    int err_raised = 0;
+    if (!is_valid_uri(value) && strncmp(name, "URI", 3) == 0) {
```

Comment:
Why do you have `starts_with` but here you use `strncmp` directly?  I think that shows that the first one should be, at most, a  macro.